### DISCUSSION TITLE
Плагин не учитывает скидки или купоны (промокоды), что приводит к неправильному расчету наложенного платежа.

### DISCUSSION THREAD

**Discussion by Dustor** [cdek-it/wordpress#43] [Ideas]
Соответственно, приходится вручную проверять каждый заказ и модифицировать уже на сайте СДЭК. Что очень неудобно, иногда приводит к ошибкам, и покупатели ругаются. Проблема легко повторяется, достаточно на сайте включить использование купонов, это стандартный функционал WooCommerce.

**Comment by vermorag** [comment]
Добрый день. Данной функциональности действительно нет в плагине. Заведу задачу на доработку - учтем в будущих версиях.

**Reply by Dustor** [reply]
На самом деле вроде как все просто, вы используете product->get_price(), а этот метод дает изначальную цену продукта. А надо наверно использовать item->get_total
Я не настоящий программист, но по крайней мере. я так себе изменил в коде плагина СДЭК в CreateOrderAction.php: 
вызовы product->get_price() заменил на $item->get_total() / max( 1, $item->get_quantity() )

Для моих целей эта замена работает, учитывает и купоны, и скидку по количеству (сторонний плагин).
